lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php

```diff
@@ -90,21 +88,18 @@ class TransportBuilder
 
     /**
      * @param FactoryInterface $templateFactory
-     * @param MessageInterface $message
+     * @param MessageInterfaceFactory $messageFactory
      * @param SenderResolverInterface $senderResolver
-     * @param ObjectManagerInterface $objectManager
      * @param TransportInterfaceFactory $mailTransportFactory
      */
     public function __construct(
         FactoryInterface $templateFactory,
-        MessageInterface $message,
+        MessageInterfaceFactory $messageFactory,
```

Comment:
I think that this should be updated so that it does not break BC. I would recommend leaving the `MessageInterface $message,` and `ObjectManagerInterface $objectManager,`  but marking them as deprecated and also unused parameters in the __construct doc block. Then adding a new optional injection of the factory. You can find more information on this approach in http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter

Reply:
You can use `@SuppressWarnings(PHPMD.UnusedFormalParameter)` on the doc block to mark the fact you are not assigning `$this->message` and `$this->objectManager` anymore but keeping them in the constructor.

Reply:
Actually after further reading I think you should continue to assign the older parameters but mark them as deprecated. "Mark properties with the @deprecated tag instead of removing or renaming them. Continue storing the value in the property to preserve the old functionality."